webui/react/src/stores/workspaces.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { createContext, PropsWithChildren, useCallback, useContext, useState } from 'react';
+
+import { getWorkspaces } from 'services/api';
+import { Workspace } from 'types';
+import handleError from 'utils/error';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+type WorkspacesContext = {
+  updateWorkspaces: (fn: (ws: Loadable<Workspace[]>) => Loadable<Workspace[]>) => void;
+  workspaces: Loadable<Workspace[]>;
```

Comment:
I didn't add functions to modify workspaces for now but when we do we should rewrite this to use a dictionary instead of an array. Since this is causing performance problems on latest-master I just wanted to get this in place for the experiment listing instead of going through the application and converting all places that use workspaces.